beacon-chain/sync/initial-sync/round_robin.go

```diff
@@ -159,47 +159,46 @@ func (s *Service) processFetchedDataRegSync(
 	ctx context.Context, genesis time.Time, startSlot primitives.Slot, data *blocksQueueFetchedData) {
 	defer s.updatePeerScorerStats(data.pid, startSlot)
 
-	blockReceiver := s.cfg.Chain.ReceiveBlock
-	invalidBlocks := 0
-	blksWithoutParentCount := 0
+	bwb, err := validUnprocessed(ctx, data.bwb, s.cfg.Chain.HeadSlot(), s.isProcessedBlock)
+	if err != nil {
+		log.WithError(err).Debug("batch did not contain a valid sequence of unprocessed blocks")
+		return
+	}
+	if len(bwb) == 0 {
+		return
+	}
+	bv := newBlobBatchVerifier(s.newBlobVerifier)
+	avs := das.NewLazilyPersistentStore(s.cfg.BlobStorage, bv)
+	batchFields := logrus.Fields{
+		"firstSlot":        data.bwb[0].Block.Block().Slot(),
+		"firstUnprocessed": bwb[0].Block.Block().Slot(),
+	}
 	for _, b := range data.bwb {
-		if len(b.Blobs) > 0 {
-			verified, err := verification.BlobSidecarSliceNoop(b.Blobs)
-			if err != nil {
-				log.WithField("root", b.Block.Root()).WithError(err).Error("blobs failed verification")
-				continue
-			}
-			for i := range verified {
-				if err := s.cfg.BlobStorage.Save(verified[i]); err != nil {
-					log.WithError(err).Warn("Failed to save blob sidecar")
-				}
-			}
+		if err := avs.Persist(s.clock.CurrentSlot(), b.Blobs...); err != nil {
```

Comment:
There's a potential DA memory exploit here if we persist without ensuring the block is valid. Here's a summary:
- First, we save blobs to memory in the sync package.
- Next, in the blockchain package, we verify the blob's validity by retrieving it from the cache. We then check if the blob and block are aligned, including kzg checks. Regardless of whether they pass or fail, we always delete the blob from the cache.
- We verify the blob after verifying the block.
- It's important to note that someone could provide blobs referencing a bad block. In this case, the blobs would remain stuck in memory.

To fix this, we have a few options:
- Implement `defer s.cache.delete(key)` at the `ReceiveBlock` level to ensure the blob is always cleared when the function ends
- Add a TTL to the blob cache
- Avoid persisting blobs to the cache unless the block is confirmed valid. This change might be less elegant, and I'm not entirely convinced it's the best approach

Reply:
Correction: The scope of AVS is per batch. This is not an issue